### PR TITLE
Add convert bits

### PR DESCRIPTION
**Pull request by clarkmoody**
Imports the `convert_bits` function from `rust-bech32-bitcoin`.

### CODE REVIEW

**Comment by clarkmoody** [comment]
@sgeisler @apoelstra Any comments on this one?

**Comment by apoelstra** [comment]
Would it make sense to add `BitcoinConversionError` to the top-level `Error` enum?

**Comment by clarkmoody** [comment]
Yep. We can reuse `Error::InvalidData(u8)` and add `InvalidPadding` to the existing enum.

**Comment by apoelstra** [comment]
Great, if you can do that then ACK from me.

**Comment by apoelstra** [comment]
ACK

**Comment by clarkmoody** [comment]
Resolves #3

**Comment by sgeisler** [comment]
I'd like to change the type of `data` to `&[u8]`, so the signature becomes  `fn convert_bits(data: &[u8], from: u32, to: u32, pad: bool)`. I will include this in #6 after rebasing on master.

File: src/lib.rs [hunk 1]
```diff
@@ -287,15 +290,63 @@ impl error::Error for Error {
...
+///
+/// # Examples
+///
+/// ```rust
+/// use bech32::convert_bits;
+/// let base5 = convert_bits(vec![0xff], 8, 5, true);
+/// assert_eq!(base5.unwrap(), vec![0x1f, 0x1c]);
+/// ```
+pub fn convert_bits(data: Vec<u8>, from: u32, to: u32, pad: bool) -> ConvertResult {
+    if from > 8 || to > 8 {
```
Comment (sgeisler) [inline src/lib.rs]: What happens if from or to is 0? From looking at the code the function could run indefinitely. Maybe it should panic instead.

Comment (apoelstra) [reply src/lib.rs]: Yeah, panicking would be better.

Comment (clarkmoody) [reply src/lib.rs]: `panic` on 0 would need a couple additional tests as well

Comment (sgeisler) [reply src/lib.rs]: opened #11 to fix this, but it would be nice if #10 could be merged first since rebasing is easier for me this way.

**Pull request merged**